sdk-core/src/main/java/co/reachfive/identity/sdk/core/MfaClient.kt

```diff
@@ -115,56 +115,77 @@ internal class MfaClient(
     }
 
     override fun startStepUp(
-        authToken: AuthToken,
+        authToken: AuthToken?,
+        stepUpToken: String?,
         authType: CredentialMfaType,
         redirectUri: String,
         scope: Collection<String>,
         success: Success<StartMfaPasswordlessResponse>,
         failure: Failure<ReachFiveError>,
-        activity: Activity,
+        activity: Activity?,
         origin: String?
     ) {
-        PkceAuthCodeFlow.generate(redirectUri).let { pkce ->
-            PkceAuthCodeFlow.storeAuthCodeFlow(pkce, activity)
-            reachFiveApi
-                .getMfaStepUpToken(
-                    authToken.authHeader,
-                    StartStepUpRequest
-                        (
-                        clientId = sdkConfig.clientId,
-                        redirectUri = redirectUri,
-                        codeChallenge = pkce.codeChallenge,
-                        codeChallengeMethod = pkce.codeChallengeMethod,
-                        scope = formatScope(scope)
+        if (authToken == null && stepUpToken == null) {
+            return failure(ReachFiveError.from("authToken and stepUpToken cannot be both null"))
+        }
+        if(stepUpToken != null) {
+            achieveStartStepUp(authType, redirectUri, stepUpToken, origin, success, failure)
+        } else {
+            PkceAuthCodeFlow.generate(redirectUri).let { pkce ->
+                val activityGuard = activity ?: return failure(ReachFiveError.from("activity must not be null"))
+                val token = authToken ?: return failure(ReachFiveError.from("auth token must not be null"))
```

Comment:
Perhaps instead define two methods with the respective signatures. Combining the different flows in a single method seems confusing to me, esp. the part where `stepUpToken` takes precedence over `authToken`.

Reply:
You mean to allow integrator to call 2 startStepUp distincts depending on the flow ? 
If it is the case, the topic has been discussed in the comments section of the ticket https://reach5.atlassian.net/browse/CA-3825?focusedCommentId=23790 and the product prefers to have one method for both flow.